src/service/action-impl.js

```diff
@@ -121,6 +126,39 @@ export class ActionService {
   }
 
   /**
+   * Installs action handler for the specified element.
+   * @param {!Element} target
+   * @param {function(!ActionInvocation)} handler
+   */
+  installActionHandler(target, handler) {
+    const debugid = target.tagName + '#' + target.id;
+    assert(this.isAmpElement_(target), 'AMP element is expected: %s', debugid);
+
+    const currentQueue = target[ACTION_QUEUE_];
+    if (currentQueue) {
+      assert(Array.isArray(currentQueue), 'Expected queue to be an array: %s',
+          debugid);
+    }
+
+    // Override queue with the handler.
+    target[ACTION_QUEUE_] = {'push': handler};
```

Comment:
> All actions are explicitly async

But our invocation isn't. Before installing this handler, all queued actions will be called at some later point (when we install it). After installing it, the action is called immediately.